channels/security/websocket.py

```diff
@@ -85,9 +85,12 @@ def match_allowed_origin(self, parsed_origin, pattern):
         ``http://exapmple.com``). After the domain there must be a port,
         but it can be omitted.
 
-        Note. This function assumes that the given origin has a schema, domain and port,
-        or only domain. Port is optional.
+        Note. This function assumes that the given origin is None, has a schema,
```

Comment:
This reads confusingly - I suggest `is either None, a schema-domain-port string, or just a domain string`